gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -260,15 +757,274 @@ gst_tensor_src_iio_finalize (GObject * object)
   //FIXME: fill this function
 }
 
+/**
+ * @brief write the string in to the file
+ * @param[in] self Tensor src IIO object
+ * @param[in] file Destination file for the data
+ * @param[in] base_dir Directory containing the file
+ * @param[in] contents Data to be written to the file
+ * @return True if write was successful, false on failure
+ */
+static gboolean
+gst_tensor_write_sysfs_string (GstTensorSrcIIO * self, const gchar * file,
+    const gchar * base_dir, const gchar * contents)
+{
+  gchar *filename = NULL;
+  gboolean ret = FALSE;
+  gint bytes_printed = 0;
+  FILE *fd = NULL;
+  GError *error = NULL;
+
+  filename = g_build_filename (base_dir, file, NULL);
+  fd = fopen (filename, "w");
+  if (fd == NULL) {
+    GST_ERROR ("Unable to open file to write %s.\n", filename);
+    goto error_free_filename;
+  }
+
+  bytes_printed = fprintf (fd, "%s", contents);
+  if (bytes_printed != strlen (contents)) {
+    GST_ERROR ("Unable to write to file %s.\n", filename);
```

Comment:
Suggestion:
Later, please consider GST_ERROR_OBJECT (self, .. ) for filtering logs with element name.

Reply:
Noted, will update in next PR.
I think would be not very clean with the helper functions as using GST_ERROR_OBJECT (self, ..) would require passing self object to all the helper functions.
Is there any better way to handle this?